### PR TITLE
Add possibility to specify ionic strength

### PR DESCRIPTION
**Pull request by andrrizzi**
Should be ready for review!

Only monovalent ions are supported right now (I've added an assert statement to check this in the code).

### CODE REVIEW

File: Yank/pipeline.py [hunk 1]
```diff
@@ -1424,6 +1457,22 @@ def _setup_system(self, system_file_path, pack, alchemical_charge,
...
+        with mmtools.utils.temporary_directory() as tmp_dir:
+            system_file_path = os.path.join(tmp_dir, 'temp_system.prmtop')
+            self._setup_system(system_file_path, *args, ignore_ionic_strength=True,
+                               save_amber_files=False)
+
+            # Count number of waters of created system.
+            system_file_path = os.path.join(tmp_dir, 'temp_system.pdb')
+            system_traj = mdtraj.load(system_file_path)
+            n_waters = 0
+            for res in system_traj.topology.residues:
```
Comment (jchodera) [inline Yank/pipeline.py]: You can probably use a generator idiom here, like:
```python
n_waters = sum(int(res.is_water) for res in system_traj.topology.residues)
```

File: docs/yamlpages/solvents.rst [hunk 2]
```diff
@@ -323,13 +323,38 @@ Specifies the negative counter ions that will be added as needed.
 No negative counter ions will be added if this option is not specified. Note that the name must match a known atom type
 in LEaP based on the parameter files you specified to load.
 
-Nonbonded Methods: ``CuttoffPeriodic``, ``Ewald``, ``PME``
+Nonbonded Methods: ``CutoffPeriodic``, ``Ewald``, ``PME``
```
Comment (jchodera) [inline docs/yamlpages/solvents.rst]: I think we want to remove ``CutoffPeriodic`` from the docs for now!

File: docs/yamlpages/solvents.rst [hunk 3]
```diff
@@ -323,13 +323,38 @@ Specifies the negative counter ions that will be added as needed.
...
+       ionic_strength: 0.0*molar
+
+The ionic strength of the ions.
+
+Both ``positive_ion`` and ``negative_ion`` must be specified with this, and only monovalent ions are supported. Note
+that this does not include the ions that are used to neutralize the periodic box.
+
+Nonbonded Methods: ``CutoffPeriodic``, ``Ewald``, ``PME``
+
+Valid Options (0 * molar) <Quantity Length> [1]_
```
Comment (jchodera) [inline docs/yamlpages/solvents.rst]: I'm confused about what this line means, and why there's a `<Quantity Length>`.

**Comment by jchodera** [comment]
Looks good to me except for the documentation bit!

File: docs/yamlpages/solvents.rst (hunk 2 above)
Comment (andrrizzi) [inline docs/yamlpages/solvents.rst]: Right! I'll take a pass.

File: docs/yamlpages/solvents.rst (hunk 3 above)
Comment (andrrizzi) [inline docs/yamlpages/solvents.rst]: Whops! Copy-paste error. I'll fix it.

The value in-between parenthesis is the default value. It's the same syntax used throughout the docs.

**Comment by andrrizzi** [comment]
Tests pass!

**Pull request merged**